src/ui/components/pages/importEmail/signContext.tsx

```diff
@@ -0,0 +1,161 @@
+import clsx from 'clsx';
+import { LedgerConnectModal } from 'ledger/connectModal';
+import { ledgerService, LedgerServiceStatus } from 'ledger/service';
+import { usePopupSelector } from 'popup/store/react';
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button } from 'ui/components/ui/buttons/Button';
+import { Modal } from 'ui/components/ui/modal/Modal';
+import Background from 'ui/services/Background';
+
+import { Login } from './login';
+import * as styles from './signWrapper.module.css';
+
+export const SignContext = createContext<{
+  openConfirmModal: () => Promise<unknown>;
+  closeConfirmModal: () => void;
+}>({
+  openConfirmModal: async () => undefined,
+  closeConfirmModal: () => undefined,
```

Comment:
I think that in general instead of passing noops as initial value it's better to make it null, but add an invariant to `useSign` hook. So in case there's no context (we're trying to call it upper in the tree) we'll get a runtime error instead of silently doing nothing.